core/txdb/snapshot.go

```diff
@@ -77,9 +80,14 @@ func storeStateSnapshot(ctx context.Context, db pg.DB, snapshot *state.Snapshot,
 		INSERT INTO snapshots (height, data) VALUES($1, $2)
 		ON CONFLICT (height) DO UPDATE SET data = $2
 	`
-
 	_, err = db.Exec(ctx, insertQ, blockHeight, b)
-	return errors.Wrap(err, "writing state snapshot to database")
+	if err != nil {
+		return errors.Wrap(err, "writing state snapshot to database")
+	}
+
+	const deleteQ = `DELETE FROM snapshots WHERE timestamp < $1`
```

Comment:
Would it be safer to do the time arithmetic in sql? Doing the subtraction in Go, if a cored has a misconfigured clock, it can accidentally delete all the snapshots. A cored clock that badly out of sync would cause other problems, for sure, but at least maybe we can avoid this one.